snekbox/memfs.py

```diff
@@ -150,21 +158,21 @@ def files(
                     log.info(f"Skipping {file.name!r} as it has not been modified")
                     continue
 
-            if count > limit:
+            if added > limit:
                 log.info(f"Max attachments {limit} reached, skipping remaining files")
                 break
 
-            if file.is_file():
-                count += 1
-                log.info(f"Found valid file for upload {file.name!r}")
-                yield FileAttachment.from_path(file, relative_to=self.output)
+            added += 1
+            log.info(f"Found valid file for upload {file.name!r}")
+            yield FileAttachment.from_path(file, relative_to=self.output)
 
     def files_list(
         self,
         limit: int,
         pattern: str,
         exclude_files: dict[Path, float] | None = None,
         preload_dict: bool = False,
+        timeout: float | None = None,
```

Comment:
Thoughts on using `float("inf")` (aka `math.inf`) instead of `None` to specify no timeout? Maybe a bit odd at first glance, but I think it makes sense, and it does avoid dealing with `None` checks.

Reply:
I feel like the None is still useful on performance grounds since we currently bypass the `time.monotic` call and comparison when `timeout` is None.